### PR TITLE
db/datagrid-new-api-options

### PR DESCRIPTION
**Pull request by karolkolodziej**
- [x] show/hide
- [x] header format
- [x] cell format
- [x] test
- [x] docklet
- [x] editing cell format

### CODE REVIEW

File: ts/DataGrid/DataGrid.ts [hunk 1]
```diff
@@ -397,7 +400,25 @@ class DataGrid {
...
+     * column `show` option.
+     * @internal
+     */
+    private getColumnsToDisplay(): Array<string> {
+        const columnsOptions = this.options.columns,
+            tableColumns = this.dataTable.getColumnNames();
+
+        return tableColumns.filter(function (columnName): boolean {
+            const column = columnsOptions[columnName];
+            if (column && defined(column.show)) {
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid.ts:413]: Do we need defined here?

File: ts/DataGrid/DataGrid.ts [hunk 2]
```diff
@@ -397,7 +400,25 @@ class DataGrid {
...
-     * Determs whether a column is editable or not.
+     * Check which columns should be displayed based on the individual
+     * column `show` option.
+     * @internal
+     */
+    private getColumnsToDisplay(): Array<string> {
+        const columnsOptions = this.options.columns,
+            tableColumns = this.dataTable.getColumnNames();
+
+        return tableColumns.filter(function (columnName): boolean {
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid.ts:411]: AFAIK we should not use `.filter`, `.map` etc functions due to performance reasons.

File: ts/DataGrid/DataGrid.ts [hunk 3]
```diff
@@ -517,13 +538,15 @@ class DataGrid {
 
 
             for (let k = 0; k < columnsInPresentationOrder.length; k++) {
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid.ts:540]: Would be better to replace it with `let k = 0; kEnd = columnsInPresentationOrder.length; k < kEnd` - performance reasons.

File: ts/DataGrid/DataGrid.ts [hunk 4]
```diff
@@ -784,6 +818,60 @@ class DataGrid {
...
+     * @internal
+     *
+     * @param columnName
+     * Column name to format.
+     */
+    private formatHeaderCell(columnName: string): string {
+        const options = this.options,
+            columnOptions = options.columns[columnName],
+            headerFormat = columnOptions && columnOptions.headerFormat,
+            ctx = { text: columnName };
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid.ts:833]: The variable is declared for use it once, I suggest to remove it from here.

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

**Review by pawellysy — APPROVED** [review]

File: ts/DataGrid/DataGrid.ts (hunk 1 above)
Comment (karolkolodziej) [reply ts/DataGrid/DataGrid.ts:413]: Yes, we need it ;) 
False is a valid value here, thus we have to use `define`.

File: ts/DataGrid/DataGrid.ts (hunk 2 above)
Comment (karolkolodziej) [reply ts/DataGrid/DataGrid.ts:411]: @bre1470 ?

**Review by goransle — APPROVED** [review]
Approved pending other requested changes

**Pull request merged**